source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -132,6 +142,38 @@ Http::FilterHeadersStatus FaultFilter::decodeHeaders(Http::HeaderMap& headers, b
   return Http::FilterHeadersStatus::Continue;
 }
 
+void FaultFilter::maybeSetupResponseRateLimit() {
+  if (!fault_settings_->responseRateLimit().has_value()) {
+    return;
+  }
+
+  // TODO(mattklein123): Allow runtime override via downstream cluster similar to the other keys.
+  if (!config_->runtime().snapshot().featureEnabled(
+          RuntimeKeys::get().ResponseRateLimitKey,
+          fault_settings_->responseRateLimit().value().percentage_)) {
+    return;
+  }
+
+  incActiveFaults();
```

Comment:
The way the active faults work is that it's for the entire request (all types of faults), so it isn't the case that short lived ones will starve.

IMO it's good to have a general catch-all/control for monitoring all faults, though I agree having per type stats might be additionally useful. The global counters/controls are in place so we can safely do edge fault injection effectively.

WDYT about more comments around how active faults are accounted for and a TODO for per-fault active counters and we see if anyone asks for them?

Reply:
I looked again and I have a comment in `FaultFilter::incActiveFaults()`, LMK where else I can put a comment about this?

Reply:
The order doesn't matter here, because each request effectively will only use a single active fault for accounting purposes. Your other point is completely valid around starvation but that would be across multiple requests, does that make sense? I think I will add more comments about this and maybe rename to `maybeIncActiveFaults()`? WDYT?